internal/settings/settings_test.go

```diff
@@ -40,3 +43,40 @@ func TestDecodeOptions_success(t *testing.T) {
 		t.Fatalf("options mismatch: %s", diff)
 	}
 }
+
+func TestValidate_IgnoreDirectoryNames_error(t *testing.T) {
+	tables := []struct {
+		input  string
+		result string
+	}{
+		{datadir.DataDirName, `cannot ignore data directory ".terraform"`},
+		{filepath.Join("path", "path"), fmt.Sprintf(`expected directory name, got a path: %q`, filepath.Join("path", "path"))},
+	}
+
+	for _, table := range tables {
+		out, err := DecodeOptions(map[string]interface{}{
+			"IgnoreDirectoryNames": []string{table.input},
```

Comment:
```suggestion
			"ignoreDirectoryNames": []string{table.input},
```
We may want to use the field tag names, to exercise the mapping in the test too.